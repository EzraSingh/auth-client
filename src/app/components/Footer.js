import React from 'react'

const Footer = () => {
    return(
        <footer>
            By 
            <a href="https://github.com/EzraSingh" target="_BLANK" rel="noopener noreferrer">
                Ezra Singh,
            </a>
            browse the source on
            <a href="https://github.com/EzraSingh/auth-client" target="_BLANK" rel="noopener noreferrer">
                <i className="fab fa-github"/>
            </a>
        </footer>
    )
}

export default Footer