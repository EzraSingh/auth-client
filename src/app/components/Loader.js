import React from 'react'

const Loader = () => {
    return <div className='sk-rotating-plane'/>
}

export default Loader