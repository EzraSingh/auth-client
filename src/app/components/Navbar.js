import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { NavLink } from 'react-router-dom'
import { connect } from 'react-redux'

/** Ruute template for registering routes within NavBar  */
function Route(name, href, auth, hideOnAuth){
    this.name = name
    this.href = href
    this.auth = auth
    this.hideOnAuth = hideOnAuth
}

class Navbar extends Component{
    static propTypes = {
        isLoggedIn: PropTypes.bool.isRequired,
        avatar: PropTypes.string,
        name: PropTypes.string
    }
    static routes = [
        new Route('Logout', '/logout', true)
    ]
    /** Filter routes that require authentication */
    generateRoutes(){
        if(!this.props.isLoggedIn){
            // Filter routes that do not require auth
            return Navbar.routes.filter((route) => !route.auth)
        }
        else{
            // Filter routes that require auth and are not flagged to hide on auth
            return Navbar.routes.filter((route) => route.auth && !route.hideOnAuth)
        }
    }
    renderAvatar = () =>{
        return(
            <figure className="avatar animated rotateIn delay-2">
                <NavLink to="/profile">
                    <img 
                        src={this.props.avatar || process.env.REACT_APP_DEFAULT_AVATAR} 
                        alt={this.props.name}
                    />
                </NavLink>
            </figure>
        )
    }
    render(){
        return(
            <nav className="flex navbar animated slideInDown">
                {this.generateRoutes().map((route) => (
                    <NavLink className="button button-outline" to={route.href} key={route.href}>
                        {route.name}
                    </NavLink>
                ))}
                {this.props.isLoggedIn && this.renderAvatar()}
            </nav>
        )
    }
}

const mapStateToProps = state => {
    return { ...state.auth, avatar: state.user.profile.avatar, name: state.user.profile.name }
}

export { Navbar as NavbarComponent }
export default connect(mapStateToProps)(Navbar)