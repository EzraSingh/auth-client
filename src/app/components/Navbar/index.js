import React, { Fragment } from 'react'
import { default as Navbar } from './containers/MenuContainer'

const withNav = Component => props => {
    return(
        <Fragment>
            <Navbar/>
            <div className="navbar-container">
                <Component {...props}/>
            </div>
        </Fragment>
    )
}

export default Navbar
export { withNav }