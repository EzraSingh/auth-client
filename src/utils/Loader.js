import React from 'react'

const Loader = () => {
    return(
        <em>Loading...</em>
    )
}

export default Loader