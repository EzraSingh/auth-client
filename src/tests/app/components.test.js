import React from 'react'
import { shallow } from 'enzyme'
import Footer from 'app/components/Footer'
import Loader from 'app/components/Loader'
import Navbar, { withNav } from 'app/components/Navbar'

describe('app components', () =>{
    describe('<Footer />', () => {
        it('should render', () => {
            const component = shallow(<Footer/>)
            expect(component).toMatchSnapshot()
        })
    })
    describe('<Loader />', () => {
        it('should render', () => {
            const component = shallow(<Loader/>)
            expect(component).toMatchSnapshot()
        })
    })
    describe('<Navbar />', () => {
        it('should render', () => {
            const component = shallow(<Navbar/>)
            expect(component).toMatchSnapshot()
        })
        describe('withNav HOC', () => {
            const MockComponent = () => <div id="MOCK-CONTENT"/>
            const WrappedComponent = withNav(MockComponent)
            it('should wrap component with Navbar and container', () => {
                const component = shallow(<WrappedComponent/>)
                const container = component.find('div.navbar-container')
                expect(component.children().first().equals(<Navbar/>)).toBe(true)
                expect(container.contains(<MockComponent/>)).toBe(true)
            })
        })
    })
})