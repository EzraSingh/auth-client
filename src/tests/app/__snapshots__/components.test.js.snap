// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app components <Footer /> should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "By",
        <a
          href="https://github.com/EzraSingh"
          rel="noopener noreferrer"
          target="_BLANK"
        >
          Ezra Singh,
        </a>,
        "browse the source on",
        <a
          href="https://github.com/EzraSingh/auth-client"
          rel="noopener noreferrer"
          target="_BLANK"
        >
          <i
            className="fab fa-github"
          />
        </a>,
      ],
    },
    "ref": null,
    "rendered": Array [
      "By",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Ezra Singh,",
          "href": "https://github.com/EzraSingh",
          "rel": "noopener noreferrer",
          "target": "_BLANK",
        },
        "ref": null,
        "rendered": "Ezra Singh,",
        "type": "a",
      },
      "browse the source on",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="fab fa-github"
          />,
          "href": "https://github.com/EzraSingh/auth-client",
          "rel": "noopener noreferrer",
          "target": "_BLANK",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fab fa-github",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "a",
      },
    ],
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "By",
          <a
            href="https://github.com/EzraSingh"
            rel="noopener noreferrer"
            target="_BLANK"
          >
            Ezra Singh,
          </a>,
          "browse the source on",
          <a
            href="https://github.com/EzraSingh/auth-client"
            rel="noopener noreferrer"
            target="_BLANK"
          >
            <i
              className="fab fa-github"
            />
          </a>,
        ],
      },
      "ref": null,
      "rendered": Array [
        "By",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Ezra Singh,",
            "href": "https://github.com/EzraSingh",
            "rel": "noopener noreferrer",
            "target": "_BLANK",
          },
          "ref": null,
          "rendered": "Ezra Singh,",
          "type": "a",
        },
        "browse the source on",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="fab fa-github"
            />,
            "href": "https://github.com/EzraSingh/auth-client",
            "rel": "noopener noreferrer",
            "target": "_BLANK",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fab fa-github",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "a",
        },
      ],
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`app components <Loader /> should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "sk-rotating-plane",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "sk-rotating-plane",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`app components <Navbar /> should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(MenuContainer) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
